tests/components/switchbot_cloud/test_cover.py

```diff
@@ -0,0 +1,508 @@
+    await configure_integration(hass)
+    state = hass.states.get(cover_id)
+    assert state.state == STATE_UNKNOWN
+
+
+async def test_cover_set_attributes_position_is_none_for_blind_tilt(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test cover set_attributes position is none for blind_tilt."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Blind Tilt",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    cover_id = "cover.cover_1"
+    mock_get_status.side_effect = [{"direction": "up"}, {"direction": "up"}]
+    await configure_integration(hass)
+    state = hass.states.get(cover_id)
+    assert state.state == STATE_UNKNOWN
+
+
+async def test_cover_set_attributes_coordinator_is_none(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test cover set_attributes coordinator is none."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Roller Shade",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    cover_id = "cover.cover_1"
+    mock_get_status.return_value = None
+    await configure_integration(hass)
+    state = hass.states.get(cover_id)
+    assert state.state == STATE_UNKNOWN
+
+
+async def test_cover_set_attributes_coordinator_is_none_blind_tilt(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test cover set attributes_coordinator is none blind_tilt."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Blind Tilt",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    cover_id = "cover.cover_1"
+    mock_get_status.return_value = None
+    await configure_integration(hass)
+    state = hass.states.get(cover_id)
+    assert state.state == STATE_UNKNOWN
+
+
+async def test_curtain_features(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test curtain features."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="cover-id-1",
+            deviceName="cover-1",
+            deviceType="Curtain",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+
+    mock_get_status.side_effect = [
+        {
+            "slidePosition": 95,
+        },
+        {
+            "slidePosition": 95,
+        },
+    ]
+    entry = await configure_integration(hass)
+    assert entry.state is ConfigEntryState.LOADED
+
+    cover_id = "cover.cover_1"
+    with (
+        patch.object(SwitchBotAPI, "send_command") as mock_send_command,
+        patch("asyncio.sleep", AsyncMock()),
```

Comment:
Let's make the sleep time a constant and patch the constant instead

Reply:
done